backend/routes/validate.py

```diff
@@ -0,0 +1,161 @@
+"""Authentication validation endpoints."""
+
+import httpx
+from fastapi import APIRouter, Depends, HTTPException, Request, status
+from llama_stack.distribution.server.auth_providers import (
+    AuthRequest,
+    AuthRequestContext,
+    AuthResponse,
+    User,
+)
+from sqlalchemy.ext.asyncio import AsyncSession
+
+from ..api.llamastack import (
+    get_sa_token,
+    get_user_headers_from_request,
+    token_to_auth_header,
+)
+from ..database import get_db
+from ..routes.users import get_user_from_headers
+
+router = APIRouter(prefix="/validate", tags=["validate"])
+
+SAR_VALIDATION_URL = "http://localhost:8887/validate-token"
```

Comment:
These URLs will have to be ready from env vars when deployed right ?

Reply:
probably not, at least in the code's current state. this is an oauth-proxy validation endpoint... specifically -
https://github.com/rh-ai-kickstart/ai-virtual-agent/blob/508bdb32bd6dbc403773d129058919a0246daade/deploy/helm/ai-virtual-assistant/templates/deployment.yaml#L84